lib/customValidators.js

```diff
@@ -239,9 +196,74 @@ function isSrvrSecProperArray(schemaType, specialSecTypes, secObj, secName) {
   return true;
 }
 
+/**
+ * Validates if parameters specified in the channel have corresponding parameters object defined and if name does not contain url parameters
+ * 
+ * @private
+ * @param  {Object} parsedJSON parsed AsyncAPI document
+ * @param  {String} asyncapiYAMLorJSON AsyncAPI document in string
+ * @param  {String} initialFormat information of the document was oryginally JSON or YAML
+ * @returns {Boolean} true in case the document is valid, otherwise throws ParserError
+ */
+function validateChannels(parsedJSON, asyncapiYAMLorJSON, initialFormat) {
+  const chnls = parsedJSON.channels;
+  if (!chnls) return true;
+
+  const chnlsMap = new Map(Object.entries(chnls));
+  const notProvidedParams = new Map();
+  const invalidChannelName = new Map();
+
+  chnlsMap.forEach((val, key) => {
+    const variables = parseUrlVariables(key);
+    const notProvidedChannelParams = notProvidedParams.get(tilde(key));
+    const parameters = parseUrlParameters(key);
```

Comment:
maybe rename `parameters` to `queryParameters` and `parseUrlParameters` to `parseUrlQueryParameters`?

this will make code slightly clearer. Problem with current naming is that this function validates, query params and also if variables have params object. So you see "parameters" in 2 different contexts with totally different meaning. This is why I think we need to make super clear that it is `query param` here